syntax = "proto3";

package proto;

message Tag {
    int32 tag_id = 1;
    repeated float pose_R = 2;
    repeated float pose_t = 3;
}

message AprilTags {
    string camera_name = 1;
    int32 image_id = 2;
    int64 timestamp = 3;
    repeated Tag tags = 4;
}

message Corner {
    float x = 1;
    float y = 2;
}

message TagCorners {
    int32 id = 1;
    repeated Corner corners = 2;
}

message RawAprilTagCorners {
    string camera_name = 1;
    int32 image_id = 2;
    int64 timestamp = 3;
    repeated TagCorners tags = 4;
}
